client/src/push-content.ts

```diff
@@ -1,7 +1,33 @@
 import vscode from 'vscode'
-import { expect } from './utils'
+import { expect, getErrorDiagnosticsBySource } from './utils'
 import { GitExtension, GitErrorCodes, CommitOptions, Repository } from './git-api/git'
 
+export enum DiagnosticSource {
+  xml = 'xml',
+  cnxml = 'cnxml'
+}
+
+export const PushValidationModal = {
+  cnxmlErrorMsg: 'There are outstanding validation errors that must be resolved before pushing is allowed.',
+  xmlErrorMsg: 'There are outstanding schema errors. Are you sure you want to push these changes?',
+  xmlErrorIgnoreItem: 'Yes, I know more than you'
+}
+
+export const canPush = async (errorsBySource: Map<string, Array<[vscode.Uri, vscode.Diagnostic]>>): Promise<boolean> => {
+  if (errorsBySource.has(DiagnosticSource.cnxml)) {
+    void vscode.window.showErrorMessage(PushValidationModal.cnxmlErrorMsg, { modal: true })
+    return false
+  } else if (errorsBySource.has(DiagnosticSource.xml)) {
```

Comment:
Subjective: it looks like the parent `if` is being used as a guard, so this else if could also be a guard ( via dropping the `else`)